packages/xo-web/src/common/xo/index.js

```
@@ -1457,6 +1440,36 @@ export const importVms = (vms, sr) =>
     )
   ).then(ids => ids.filter(_ => _ !== undefined))
 
+const importDisk = ({ description, file, name, type, vmdkData }, sr) =>
+  _call('disk.import', {
+    description,
+    name,
+    sr: resolveId(sr),
+    type,
+    vmdkData,
+  }).then(({ $sendTo }) =>
+    post($sendTo, file)
+      .then(async res => {
+        if (res.status !== 200) {
+          throw res.status
+        }
+        success(_('diskImportSuccess'), name)
+        await res.json().then(body => body.result)
+      })
+      .catch(err => {
+        error(_('diskImportFailed'), err)
```

Rethrow the error after that. We've had the same issue with `importVm`: we should really avoid swallowing errors as we may want to handle them higher in the stack.